chain/app/db_app.go

```diff
@@ -101,6 +138,7 @@ func (app *dbBandApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseB
 	res = app.bandApp.BeginBlock(req)
 	// Begin transaction
 	app.dbBand.BeginTransaction()
+	app.dbBand.SetContext(app.DeliverContext)
```

Comment:
Let's also clear this after EndBlock. Also should we make it readonly. Like creating a cache on top and discard after.